File: python/3sumclosest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        i=0
        j=len(nums)-1
        
        diff=float('inf')
        
        for k in range(len(nums)-2):
            i=k+1
            j=len(nums)-1
            while(i<j):
                currSum=nums[i]+nums[j]+nums[k]
                if(abs(target-currSum)<diff):
                    diff=abs(target-currSum)
                    v=currSum
                if(currSum==target):
                    return(currSum)
                elif(currSum<target):
                    i+=1
                else:
                    j-=1
        return(v)

File: java/3sumCLosest.java
int threeSumClosest(int[] numbers, int target) {
        
        Arrays.sort(numbers);
        
        int diff=Integer.MAX_VALUE;
        int closest=Integer.MAX_VALUE;
        
        for(int k=0;k<numbers.length-2;k++){
        
        int left=k+1;
        int right=numbers.length-1;
        
        while(left<right){
            
            int sum=numbers[k]+numbers[left]+numbers[right];
            
            if(Math.abs(sum-target)<diff){
                diff=Math.abs(sum-target);
                closest=sum;
            }
            
            if(sum==target){

                return sum;
            }
            
            else if(sum<target){
                
                left+=1;
            }
            
            else if(sum>target){
                right-=1;
            }
            
        }
        }
        
        return closest;
    }